pandas/core/dtypes/cast.py

```diff
@@ -1505,10 +1505,20 @@ def cast_scalar_to_array(shape, value, dtype: Optional[DtypeObj] = None) -> np.n
 
     """
     if dtype is None:
-        dtype, fill_value = infer_dtype_from_scalar(value)
+        dtype, fill_value = infer_dtype_from_scalar(value, pandas_dtype=True)
     else:
         fill_value = value
 
+    # TODO: Update this function to add support for 3rd party extension types
+    # Issue #34959
+    if is_extension_array_dtype(dtype):
+        if isinstance(shape, int):
+            shape = (shape, 1)
+        return [
+            construct_1d_arraylike_from_scalar(value, shape[0], dtype)
```

Comment:
Can you update the docstring about this new return type?

Reply:
Also, why is it actually needed to return a list if the specified shape is 1D?

Reply:
For some reason I thought that wouldn't work, but it looks like it does. Thanks for the suggestion
Also added info in the docstring
Also added constructor test

Reply:
It's a list of 1d extension arrays, ie [PeriodArray, PeriodArray] (I updated the docstring to reflect this)